repositories.go

```diff
@@ -162,7 +164,13 @@ func (s *RepositoriesService) Archive(pid interface{}, opt *ArchiveOptions, opti
 	}
 
 	var b bytes.Buffer
-	resp, err := s.client.Do(req, &b)
+	var resp *Response
+	if opt != nil && opt.Writer != nil {
+		resp, err = s.client.Do(req, opt.Writer)
+	} else {
+		resp, err = s.client.Do(req, &b)
+	}
```

Comment:
I get what you are trying to do, but I don't think this is the best way to implement this. Let me think a little about this one...

Reply:
I want to stream the content of the repository directly from a micro-service to another micro-service, via HTTP with `io.Copy`.

To do this, I propose this way of doing this, avoiding any breakage of the current method.

Is that clear for you?

Reply:
It's perfectly clear to me, but I don't like this particular solution. Let me add an alternative solution for this instead. I'll push it as additional commit to this PR so we can discuss if it will work for you.